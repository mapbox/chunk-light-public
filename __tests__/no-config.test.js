'use strict';

const path = require('path');
const tempy = require('tempy');
const fs = require('fs');
const dirTree = require('directory-tree');
const promisify = require('util.promisify');

const { copy } = require('../lib/utils/auto-copy');
const removePath = require('./test-utils/remove-path');
const bootstrapReactRepo = require('./test-utils/bootstrap-react-repo');
const commandBuild = require('./test-utils/command-build');

jest.setTimeout(15 * 1000);

describe('No Config Test', () => {
  const tempDir = tempy.directory();

  // give `urc.rootDirectory` a stable value
  const dirPath = path.join(tempDir, 'test');

  beforeAll(() =>
    promisify(fs.mkdir)(dirPath)
      .then(() => bootstrapReactRepo(dirPath))
      .then(() =>
        copy({
          sourceDir: path.join(__dirname, 'fixtures', 'no-config'),
          destDir: dirPath,
          sourceGlob: ['**/*', '.env*'] // need to add `.env*` to include dotfiles.
        })
      ));

  test('builds the correct output without any configuration', () => {
    const result = commandBuild([], { cwd: dirPath });

    const tree = removePath({
      object: dirTree(path.join(dirPath, '_underreact-site')),
      path: dirPath,
      replaceWith: '<TEMP_DIR>'
    });

    expect(tree).toMatchSnapshot();
    expect(result.status).toBe(0);
  });

  test('builds the correct output with development mode', () => {
    const result = commandBuild(['--mode=development'], { cwd: dirPath });

    const tree = removePath({
      object: dirTree(path.join(dirPath, '_underreact-site')),
      path: dirPath,
      replaceWith: '<TEMP_DIR>'
    });

    expect(tree).toMatchSnapshot();
    expect(result.status).toBe(0);
  });

  test('exits with statusCode 1 when config is not found', () => {
    const result = commandBuild(
      [`--config=${path.join(dirPath, 'not-exists.config.js')}`],
      {
        // ignore the stdio as it is expected it will log to console.error and
        // it might mistakingly look like tests are failing
        stdio: 'ignore',
        cwd: dirPath
      }
    );
    expect(result.status).toBe(1);
  });
});
