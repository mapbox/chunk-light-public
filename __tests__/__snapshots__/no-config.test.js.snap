// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Config Test builds the correct output with development mode 1`] = `
Object {
  "children": Array [
    Object {
      "extension": ".html",
      "name": "index.html",
      "path": "<TEMP_DIR>/_underreact-site/index.html",
      "size": 2256,
      "type": "file",
    },
    Object {
      "children": Array [
        Object {
          "extension": ".js",
          "name": "main.chunk.js",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/main.chunk.js",
          "size": 4392,
          "type": "file",
        },
        Object {
          "extension": ".map",
          "name": "main.chunk.js.map",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/main.chunk.js.map",
          "size": 685,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "polyfill.chunk.js",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/polyfill.chunk.js",
          "size": 97853,
          "type": "file",
        },
        Object {
          "extension": ".map",
          "name": "polyfill.chunk.js.map",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/polyfill.chunk.js.map",
          "size": 67028,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "runtime.js",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/runtime.js",
          "size": 6241,
          "type": "file",
        },
        Object {
          "extension": ".map",
          "name": "runtime.js.map",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/runtime.js.map",
          "size": 6135,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "vendor.chunk.js",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/vendor.chunk.js",
          "size": 1353,
          "type": "file",
        },
        Object {
          "extension": ".map",
          "name": "vendor.chunk.js.map",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/vendor.chunk.js.map",
          "size": 287,
          "type": "file",
        },
      ],
      "name": "underreact-assets",
      "path": "<TEMP_DIR>/_underreact-site/underreact-assets",
      "size": 183974,
      "type": "directory",
    },
  ],
  "name": "_underreact-site",
  "path": "<TEMP_DIR>/_underreact-site",
  "size": 186230,
  "type": "directory",
}
`;

exports[`No Config Test builds the correct output without any configuration 1`] = `
Object {
  "children": Array [
    Object {
      "extension": ".html",
      "name": "index.html",
      "path": "<TEMP_DIR>/_underreact-site/index.html",
      "size": 2283,
      "type": "file",
    },
    Object {
      "children": Array [
        Object {
          "extension": ".js",
          "name": "main-3b9eb6636b.chunk.js",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/main-3b9eb6636b.chunk.js",
          "size": 1268,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "polyfill-52d63dac2e.chunk.js",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/polyfill-52d63dac2e.chunk.js",
          "size": 17925,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "runtime-912ed6ecb8.js",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/runtime-912ed6ecb8.js",
          "size": 1468,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "vendor-9693d4119d.chunk.js",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets/vendor-9693d4119d.chunk.js",
          "size": 182,
          "type": "file",
        },
      ],
      "name": "underreact-assets",
      "path": "<TEMP_DIR>/_underreact-site/underreact-assets",
      "size": 20843,
      "type": "directory",
    },
  ],
  "name": "_underreact-site",
  "path": "<TEMP_DIR>/_underreact-site",
  "size": 23126,
  "type": "directory",
}
`;
